### PR TITLE
fix: Unlisted token may be skipped in position Indexer after it changes to be listed

### LINKED ISSUE #1149: [BUG] Unlisted token may be skipped in position Indexer after it changes to be listed
## Describe the bug

- Missing information between issuing and Listing

## Expected behavior

- Unlisted tokens position is processed properly after they are listed.

### PR DESCRIPTION
**Pull request by purplesmoke05**
Close #1149

### CODE REVIEW

File: batch/indexer_Position_Coupon.py [hunk 1]
```diff
@@ -162,8 +156,13 @@ def __get_contract_list(self, db_session: Session):
...
                 )
-                self.token_list.append(token_contract)
-                self.token_address_list.append(token_contract.address)
+                synced_block_number = self.__get_idx_position_block_number(
+                    db_session=db_session,
+                    token_address=listed_token.token_address
+                )
+                block_from = synced_block_number + 1
+                token = self.TargetToken(token_contract, block_from)
+                self.token_list.append(token)
```
Comment (purplesmoke05) [inline batch/indexer_Position_Coupon.py:165]: Load latest synced block number of each token. 
And store the number as instance value.

File: batch/indexer_Position_Coupon.py [hunk 2]
```diff
@@ -176,22 +175,37 @@ def __get_contract_list(self, db_session: Session):
...
         self.exchange_address_list = list(set(_exchange_list_tmp))
 
-    def __sync_all(self, db_session: Session, block_from: int, block_to: int):
-        LOG.info("syncing from={}, to={}".format(block_from, block_to))
-        self.__sync_transfer(db_session, block_from, block_to)
-        self.__sync_consume(db_session, block_from, block_to)
-        self.__sync_exchange(db_session, block_from, block_to)
-        self.__sync_escrow(db_session, block_from, block_to)
+    def __sync_all(self, db_session: Session, block_to: int):
+        sync_from = self.__get_oldest_cursor(self.token_list, block_to)
```
Comment (purplesmoke05) [inline batch/indexer_Position_Coupon.py:179]: Get oldest block number among tokens synced.

File: batch/indexer_Position_Coupon.py [hunk 3]
```diff
@@ -176,22 +175,37 @@ def __get_contract_list(self, db_session: Session):
...
+        LOG.info("syncing from={}, to={}".format(sync_from, block_to))
+
+        self.__sync_transfer(db_session, block_to)
+        self.__sync_exchange(db_session, block_to)
+        self.__sync_escrow(db_session, block_to)
+
+        self.__update_cursor(block_to + 1)
```
Comment (purplesmoke05) [inline batch/indexer_Position_Coupon.py:186]: After all sync method is done, update the block number where processor should start next.

File: app/model/db/idx_position.py [hunk 4]
```diff
@@ -62,6 +62,7 @@ class IDXPositionBondBlockNumber(Base):
 
     # sequence id
     id = Column(BigInteger, primary_key=True, autoincrement=True)
+    token_address = Column(String(42), primary_key=True)
```
[the remark this answers is not in the document]
Comment (purplesmoke05) [reply app/model/db/idx_position.py:66]: ありがとうございます。コメント・FIELDS属性への追加漏れを修正いたしました。

**Review by YoshihitoAso — APPROVED** [review]

**Pull request merged**